src/components/CheckEditor/CheckProbes/ProbesList.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useMemo } from 'react';
+import { GrafanaTheme2 } from '@grafana/data';
+import { Checkbox, Label, useStyles2 } from '@grafana/ui';
+import { css } from '@emotion/css';
+
+import { Probe } from 'types';
+import { ProbeStatus } from 'components/ProbeCard/ProbeStatus';
+
+export const ProbesList = ({
+  title,
+  probes,
+  selectedProbes,
+  onSelectionChange,
+}: {
+  title: string;
+  probes: Probe[];
+  selectedProbes: number[];
+  onSelectionChange: (probes: number[]) => void;
+}) => {
+  const styles = useStyles2(getStyles);
+
+  const handleToggleAll = () => {
+    if (allProbesSelected) {
+      onSelectionChange(selectedProbes.filter((id) => !probes.some((probe) => probe.id === id)));
+      return;
+    }
+    onSelectionChange([
+      ...selectedProbes,
+      ...(probes.filter((probe) => !probe.deprecated).map((probe) => probe.id) as number[]),
+    ]);
+  };
+
+  const handleToggleProbe = (probe: Probe) => {
+    if (!probe.id || probe.deprecated) {
+      return;
+    }
+    if (selectedProbes.includes(probe.id)) {
+      onSelectionChange(selectedProbes.filter((p) => p !== probe.id));
+      return;
+    }
+    onSelectionChange([...selectedProbes, probe.id]);
+  };
+
+  const allProbesSelected = useMemo(
+    () => probes.every((probe) => selectedProbes.includes(probe.id as number)),
+    [probes, selectedProbes]
+  );
+
+  return (
+    <div className={styles.probesColumn}>
+      <div className={styles.sectionHeader}>
+        <Checkbox id={`header-${title}`} onClick={handleToggleAll} checked={allProbesSelected} />
+        <Label htmlFor={`header-${title}`} className={styles.headerLabel}>
+          {title} ({probes.length})
+        </Label>
+      </div>
+      <div className={styles.probesList}>
+        {probes.map((probe: Probe) => (
+          <div key={probe.id} className={styles.item}>
+            <Checkbox
+              id={`probe-${probe.id}`}
+              onClick={() => handleToggleProbe(probe)}
+              checked={selectedProbes.includes(probe.id as number)}
```

Comment:
We need to fix the typings for when things are in a draft versus published state at some point so we can avoid this kind of type casting 😅